src/main/java/org/json/NumberConversionUtil.java

```diff
@@ -83,6 +83,16 @@ static Number stringToNumber(final String input) throws NumberFormatException {
         throw new NumberFormatException("val ["+input+"] is not a valid number.");
     }
 
+    /**
+     * Checks if the character is a numeric digit ('0' to '9').
+     *
+     * @param c The character to be checked.
+     * @return true if the character is a numeric digit, false otherwise.
+     */
+    private static boolean isNumericChar(char c) {
+        return (c >= '0' && c <= '9');
```

Comment:
probably a stupid optimization, but switching this comparison to be:

```java
return (c <= '9' && c >= '0');
```

should provide slightly better performance. My assumption here is that in both ASCII and Unicode, there are a lot more code points that are `>'9'` than there are `<'0'`. That means that for most inputs we can fail-fast with a single comparison instead of having to do both compares.

Reply:
@adityap27 Please make this change, it won't affect the comment window duration.

Reply:
@stleary and @johnjaylward - done, updated.